packages/npm/financekit/src/index.tsx

```diff
@@ -143,99 +161,472 @@ const AccountBalance = S.Struct({
   currentBalance: CurrentBalance,
 })
 
-export async function requestAuthorization(): Promise<
-  typeof AuthorizationStatus.Type
+class FinancekitError extends S.TaggedError<FinancekitError>()(
+  'FinancekitError',
+  {
+    code: S.Literal(
+      '@candlefinance.financekit.account_balances_invalid',
+      '@candlefinance.financekit.account_balance_history_invalid',
+      '@candlefinance.financekit.accounts_invalid',
+      '@candlefinance.financekit.os_version_too_low',
+      '@candlefinance.financekit.params_invalid',
+      '@candlefinance.financekit.query_invalid',
+      '@candlefinance.financekit.transaction_history_invalid',
+      '@candlefinance.financekit.transactions_invalid',
+      '@candlefinance.financekit.unknown',
+      '@candlefinance.financekit.unknown_account_type',
+      '@candlefinance.financekit.unknown_authorization_status',
+      '@candlefinance.financekit.unknown_current_balance_type',
+      '@candlefinance.financekit.unknown_parsing_error'
+    ),
+    backingError: S.Any,
+  }
+) {}
+
+const makeError = <CODE extends String & FinancekitError['code']>(
```

Comment:
```suggestion
const makeError = <Code extends String & FinancekitError['code']>(
```

TS generics are conventionally either `A`, `T` etc or upper camel case eg `Code`